sdk/communication/Azure.Communication.CallingServer/src/Models/MediaStreaming/MediaStreamingPackageParser.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Text;
+using System.Text.Json;
+using Azure.Communication.CallingServer.Models.MediaStreaming;
+using Azure.Core;
+
+namespace Azure.Communication.CallingServer
+{
+    /// <summary>
+    /// Parser of the different packages received as part of
+    /// Media streaming.
+    /// </summary>
+    public static class MediaStreamingPackageParser
+    {
+        /// <summary>
+        /// Parsing a MediaStreaming package from BinaryData.
+        /// </summary>
+        /// <param name="json"></param>
+        /// <returns></returns>
+        public static MediaStreamingPackageBase Parse(BinaryData json)
+        {
+            return Parse(json.ToString());
+        }
+
+        /// <summary>
+        /// Parsing a MediaStreaming package from a byte array.
+        /// </summary>
+        /// <param name="receivedBytes">a UTF8 byte array.</param>
+        /// <returns></returns>
+        public static MediaStreamingPackageBase Parse(byte[] receivedBytes)
+        {
+            return Parse(Encoding.UTF8.GetString(receivedBytes));
+        }
+
+        /// <summary>
+        /// Parsing Audio packages from BinaryData.
+        /// </summary>
+        /// <param name="json"></param>
+        /// <returns></returns>
+        public static MediaStreamingPackageBase[] ParseMany(BinaryData json)
```

Comment:
As of now, we don't have the incoming data as BinaryData so we don't need this parsing.

Reply:
Isn't there a way to create BinaryData out of incoming bytes? I got this from CloudEvents that I assume are transmitted similarly to your packages, but I may be wrong.